channeldb/invoices.go

```diff
@@ -2551,21 +2571,23 @@ func cancelSingleHtlc(resolveTime time.Time, htlc *invpkg.InvoiceHTLC,
 	return nil
 }
 
-// updateHtlc aligns the state of an htlc with the given invoice state. A
-// boolean is returned if the HTLC was settled.
-func updateHtlc(resolveTime time.Time, htlc *invpkg.InvoiceHTLC,
-	invState invpkg.ContractState, setID *[32]byte) (bool, error) {
+// getUpdatedHtlcState aligns the state of an htlc with the given invoice state.
+// A boolean indicating whether the HTLCs state need to be updated, along with
+// the new state (or old state if no change is needed) is returned.
+func getUpdatedHtlcState(htlc *invpkg.InvoiceHTLC,
```

Comment:
👍 for eliminating the mutation here, will make it easier to test in the future, and go with the approach of separating figuring out what needs to be changed from actually changing it